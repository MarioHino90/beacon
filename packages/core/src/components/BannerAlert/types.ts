import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
