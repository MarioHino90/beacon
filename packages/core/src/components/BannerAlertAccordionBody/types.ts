export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
