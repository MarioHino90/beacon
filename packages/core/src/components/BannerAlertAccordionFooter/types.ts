export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
