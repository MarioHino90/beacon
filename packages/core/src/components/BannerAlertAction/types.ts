import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
