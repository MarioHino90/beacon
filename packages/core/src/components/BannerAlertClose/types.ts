import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
