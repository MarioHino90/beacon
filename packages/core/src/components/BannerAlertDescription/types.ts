export interface BannerAlertDescriptionProps
  extends React.HTMLAttributes<HTMLDivElement> {}
