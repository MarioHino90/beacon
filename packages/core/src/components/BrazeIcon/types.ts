import React from "react";

export type GenericSpotIcon<Props> = React.LazyExoticComponent<{
  (props: Props): React.ReactElement;
  displayName: string;
}>;

export interface SpotIcons {
  "add-document": GenericSpotIcon<AddDocumentProps>;
  "add-image-or-document": GenericSpotIcon<AddImageOrDocumentProps>;
  "arrow-toggle": GenericSpotIcon<ArrowToggleProps>;
  "campaign-comparison": GenericSpotIcon<CampaignComparisonProps>;
  content: GenericSpotIcon<ContentProps>;
  "email-settings": GenericSpotIcon<EmailSettingsProps>;
  "empty-table": GenericSpotIcon<AddDocumentProps>;
  "funnel-report": GenericSpotIcon<FunnelReportProps>;
  mms: GenericSpotIcon<MMSProps>;
  preview: GenericSpotIcon<PreviewProps>;
  "retention-report": GenericSpotIcon<RetentionReportProps>;
  sms: GenericSpotIcon<SMSProps>;
}

export interface BrazeIconProps extends StyledBrazeIconProps {
  /**
   * icon to be displayed to user */
  name?: keyof SpotIcons;
}

export interface StyledBrazeIconProps
  extends React.HTMLAttributes<HTMLElement> {
  /**
   * child elements to be rendered inside the component tree */
  children?: React.ReactNode | React.ReactNode[];
}

export interface AddDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface AddImageOrDocumentProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ArrowToggleProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface CampaignComparisonProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface ContentProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface FunnelReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface EmailSettingsProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface PreviewProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface MMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface RetentionReportProps
  extends React.SVGAttributes<HTMLOrSVGElement> {}

export interface SMSProps extends React.SVGAttributes<HTMLOrSVGElement> {}
