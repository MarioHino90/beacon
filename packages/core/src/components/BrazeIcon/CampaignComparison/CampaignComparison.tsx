import React from "react";
import classNames from "classnames";

import { CampaignComparisonProps } from "../types";

export const CampaignComparison = ({
  className,
  ...props
}: CampaignComparisonProps): React.ReactElement => (
  <svg
    {...props}
    className={classNames(["bcl-campaign-comparison", className])}
    width="24"
    height="24"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      x="1"
      y="8.99976"
      width="14"
      height="14"
      fill="white"
      stroke="black"
      strokeLinejoin="round"
    />
    <rect
      x="7"
      y="1"
      width="16"
      height="16"
      rx="8"
      fill="white"
      stroke="black"
      strokeLinejoin="round"
    />
    <path
      d="M7 8.99976H15V16.9998V16.9998C10.5817 16.9998 7 13.418 7 8.99976V8.99976Z"
      fill="#B5EDFF"
      stroke="black"
      strokeLinejoin="round"
    />
  </svg>
);

CampaignComparison.displayName = "CampaignComparison";
