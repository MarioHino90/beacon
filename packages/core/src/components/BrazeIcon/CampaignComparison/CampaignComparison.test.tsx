import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { CampaignComparisonProps } from "../types";

import { CampaignComparison } from "./CampaignComparison";

faker.seed(1603832397842);

describe("CampaignComparison", () => {
  let props: CampaignComparisonProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<CampaignComparison {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
