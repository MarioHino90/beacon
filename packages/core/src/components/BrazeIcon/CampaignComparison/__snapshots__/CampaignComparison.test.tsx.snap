// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignComparison should match the snapshot 1`] = `
<svg
  className="bcl-campaign-comparison impactful"
  fill="none"
  height="24"
  id="Cambridgeshire"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="white"
    height="14"
    stroke="black"
    strokeLinejoin="round"
    width="14"
    x="1"
    y="8.99976"
  />
  <rect
    fill="white"
    height="16"
    rx="8"
    stroke="black"
    strokeLinejoin="round"
    width="16"
    x="7"
    y="1"
  />
  <path
    d="M7 8.99976H15V16.9998V16.9998C10.5817 16.9998 7 13.418 7 8.99976V8.99976Z"
    fill="#B5EDFF"
    stroke="black"
    strokeLinejoin="round"
  />
</svg>
`;
